docs/architecture/adr-template.md

```diff
@@ -6,12 +6,22 @@
 
 ## Status
 
-> A decision may be "proposed" if it hasn't been agreed upon yet, or "accepted"
-> once it is agreed upon. Once the ADR has been implemented mark the ADR as
-> "implemented". If a later ADR changes or reverses a decision, it may be marked
-> as "deprecated" or "superseded" with a reference to its replacement.
-
-{Deprecated|Declined|Accepted|Implemented}
+> An architecture decision is considered "proposed" when a PR containing the ADR
+> is submitted. An ADR that is merged is automatically considered "accepted",
+> and its implementation status must be tracked via a tracking issue, milestone
+> or project board.
```

Comment:
I think I'm mildly worried about what this may mean for our ADR process. I tend to think that it's important to keep around the ADR even if the decision is _don't do this right now or ever_. It's a record of the decision that we made and what factors led to that decision. In the future if someone wants to propose something very similar to something we previously rejected, there won't be as obvious of a record of us having previously said "yeah, we do not want to do this".

Reply:
I agree with your analysis, but if I understand correctly the proposal is not to get rid of "rejected", only "proposed". That is: We might merge an ADR as rejected, or as accepted, but not as "propseed". 

(I think that was the spirit of the idea, anyway)

Reply:
@williambanfield I didn't actually explicitly think of capturing rejected ADRs, but I agree we should keep them. To elaborate on @creachadair's point, now that I've reflected on it a bit more, the main reasons for this PR were to:

1. Add some kind of link between accepted ADRs and our GitHub workflow such that readers can discuss and track implementation progress (e.g. what's the plan for [this trust metric ADR](https://github.com/tendermint/tendermint/blob/44d9e9917c3b67674833fcd8d611f18d2eb366df/docs/architecture/adr-006-trust-metric.md)? Is it still even relevant? A tracking issue would be super helpful in being able to discuss that).
2. Avoid dangling proposed ADRs with no clear path to acceptance or opportunity to discuss (e.g. [this one](https://github.com/tendermint/tendermint/blob/44d9e9917c3b67674833fcd8d611f18d2eb366df/docs/architecture/adr-007-trust-metric-usage.md)).
3. Centralize the conversations somewhat for each distinct ADR so that they're not scattered across a bunch of issues. I'm hoping this conversation centralization will make the codebase more friendly to newcomers who are trying to gain context.

I'll expand the text of the `Status` section accordingly to cater for the possibility of merging explicitly rejected ADRs.

Reply:
Ah! Thanks for clarifying, I seem to have misunderstood. 

Generally, I think this a good change. I think we'll probably spend a little bit of time figuring out what the right structure for tracking work on an ADR is since we also have the 'project' board tool that may help in this regard. 

I think that the idea of tracking what supersedes what may get a bit tricky if an ADR touches multiple areas but doesn't fully overturn a previous decision. I think it's worth experimenting with to see how it goes for us.

Reply:
It's plain text/Markdown, so it's technically a free-for-all as to what people could write in that Status section :slightly_smiling_face:

I'd imagine it's a given that some ADRs will partially override others, but there may be instances where one clearly supersedes another, and I think the "Supersedes" status should be used for those instances.

Alternatively, if we wanted, we could just have an "Updates" or "Updated by" status instead of "Supersedes" (similar to how IETF RFCs work)?